espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/SourceDocument.kt

```diff
@@ -130,34 +130,75 @@ class SourceDocument @JvmOverloads constructor(
         }
 
         val viewElement = ViewElement(view)
-        val tagName = toXmlNodeName(viewElement.className)
+        val className = viewElement.className
+        val tagName = toXmlNodeName(className)
         serializer?.startTag(NAMESPACE, tagName)
 
         // Set attributes
-        setAttribute(ViewAttributesEnum.INDEX, viewElement.index)
-        setAttribute(ViewAttributesEnum.PACKAGE, viewElement.packageName)
-        setAttribute(ViewAttributesEnum.CLASS, viewElement.className)
-        setAttribute(ViewAttributesEnum.CONTENT_DESC, viewElement.contentDescription)
-        setAttribute(ViewAttributesEnum.CHECKABLE, viewElement.isCheckable)
-        setAttribute(ViewAttributesEnum.CHECKED, viewElement.isChecked)
-        setAttribute(ViewAttributesEnum.CLICKABLE, viewElement.isClickable)
-        setAttribute(ViewAttributesEnum.ENABLED, viewElement.isEnabled)
-        setAttribute(ViewAttributesEnum.FOCUSABLE, viewElement.isFocusable)
-        setAttribute(ViewAttributesEnum.FOCUSED, viewElement.isFocused)
-        setAttribute(ViewAttributesEnum.SCROLLABLE, viewElement.isScrollable)
-        setAttribute(ViewAttributesEnum.LONG_CLICKABLE, viewElement.isLongClickable)
-        setAttribute(ViewAttributesEnum.PASSWORD, viewElement.isPassword)
-        setAttribute(ViewAttributesEnum.SELECTED, viewElement.isSelected)
-        setAttribute(ViewAttributesEnum.VISIBLE, viewElement.isVisible)
-        setAttribute(ViewAttributesEnum.BOUNDS, viewElement.bounds.toShortString())
-        val viewText = viewElement.text
-        viewText?.let {
-            setAttribute(ViewAttributesEnum.TEXT, it.rawText)
-            setAttribute(ViewAttributesEnum.HINT, it.isHint)
+        if (null == includedAttributes || includedAttributes.contains(ViewAttributesEnum.INDEX)) {
```

Comment:
How about define a variable for `null == includedAttributes`?
```kotlin
val shouldSetAllAttributes = null == includedAttributes
if (shouldSetAllAttributes || includedAttributes.contains(ViewAttributesEnum.INDEX)) {
  // ...
}
```

Reply:
I've refactored it in more functional style